### ISSUE TITLE
Send message from server to specific users

### ISSUE THREAD

**Issue by deoqc** [SocketCluster/socketcluster#234]
**tl;dr**: Possible solution:

Each client subscribe to its own `user_id`. And server publishes to each `user_id` as needed. Set middleware according.

**Is there a better approach? If not, is this a good approach or should try to model it differently?**

---

Use case:

Client emit message to server. 

For each arriving message, server/db can determine to whom (which means user_id in authToken) should it be broadcasted.

This info may change, but the correct/up-to-date use_id's are in db. So I want to broadcast this message to them.

---

Why not use regular pub/sub?

Only DB knows who belongs to each channel, and it can change. So to use pub/sub would need to send a list to client of each channel it needs to subscribe on connect. 

And update it if the DB is updated. Handling this seem too brittle.

**Comment by jondubois** [comment]
@deoqc Publishing to a channel which is named after the username/userid as you described is actually a very good and scalable approach. Channels in SC are cheap so you should use them.
